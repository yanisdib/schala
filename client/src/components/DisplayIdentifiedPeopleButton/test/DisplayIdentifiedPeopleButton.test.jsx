import { shallow } from 'enzyme';

import { DisplayIdentifiedPeopleButton } from '../..';


describe('DisplayIdentifiedPeopleButton component', () => {
    let wrapper, onButtonClick;
    const { onClick } = DisplayIdentifiedPeopleButton.defaultProps;

    beforeEach(() => {
        onButtonClick = jest.fn();
        wrapper = shallow(<DisplayIdentifiedPeopleButton onClick={onButtonClick} />);
    })

    it('renders correctly', () => expect(wrapper).toMatchSnapshot())

    it('has default props', () => expect(onClick).toBeDefined())

    it('calls onClick event', () => {
        wrapper.simulate('click');
        expect(onButtonClick).toHaveBeenCalled();
    })
})