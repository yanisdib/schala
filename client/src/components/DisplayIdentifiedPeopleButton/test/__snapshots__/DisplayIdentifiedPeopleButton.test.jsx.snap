// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayIdentifiedPeopleButton component renders correctly 1`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Styled(User) />
</styled.div>
`;
